src/cmd.ts

```diff
@@ -295,13 +300,31 @@ program
 	)
 	.action(() => void generateConfig());
 
-program
-	.command("clear-cache")
-	.description("Clear the cache of downloaded-and-rejected torrents")
-	.action(async () => {
-		await db("decision").whereNull("info_hash").del();
-		await db.destroy();
-	});
+createCommandWithSharedOptions(
+	"clear-cache",
+	"Clear the cache of downloaded-and-rejected torrents and re-search your library over a few weeks",
+).action(async (options) => {
+	validateAndSetRuntimeConfig(options);
+	const { excludeRecentSearch } = getRuntimeConfig();
+	console.log("Clearing cache, this may take a few minutes...");
+
+	await db("decision").whereNull("info_hash").del();
+	const now = Date.now();
+	const start = now - (excludeRecentSearch ?? 0);
+	const limit = ms("30 days");
+	const rows = await db("timestamp").select("searchee_id").distinct();
+	for (const row of rows) {
+		const offset = Math.floor(Math.random() * limit);
+		await db("timestamp")
+			.where({ searchee_id: row.searchee_id })
+			.update({
+				first_searched: now + offset,
+				last_searched: start + offset,
+			});
+	}
```

Comment:
I'm not really a fan of misrepresenting first and last searched in the database to achieve this goal of spreading when we have `searchLimit` that does the same thing without lying about the numbers. Is there any reason we can't rely on searchLimit here?

Reply:
We could clear the table and have the users rely on that. The timestamps would come out to same in the end but I imagine a large number of users won't do that. So I think it's better to force it since these have already been searched anyways.

Reply:
Can we force searchLimit then?

Reply:
I'm not sure of a good way to do that. Maybe checking the searchee and timestamps table and enforcing it if `x` amount of first time searches? The thing is, this only a problem for launch. So what we can do instead is just hard code a search limit for the first few weeks. Regardless, this would affect new searchees as well and if a tracker was down, it will research the same searchees the next day on that tracker only.

The more I think about it, the more I find the timestamps solution to be better as it's a direct solution to the problem. There is nothing required from the user, no code change, and guarantees the desired behavior. We can have a max search limit of 500 as a fallback incase users start with a long exclude and lower it after clearing.

In any case, we will change it to clear the entire table after a few weeks, the concern is just tens of thousands of users doing this simultaneously at launch.

Reply:
Has there been a concensus on this @mmgoodnow

Reply:
I think we can make clear-cache wipe out the timestamps completely, and just not emphasize in the migration guide that people should do it. If they are satisfied with their cross-seed catalog they don't need to do anything special. If they want to start fresh and are have the initiative to search around for how to do it, they will find the clear-cache command and run it. I think the reduced volume of users performing this action, combined with the other protections we have in place around `excludeOlder`/`excludeRecentSearch` and `delay` are probably sufficient to mitigate any load issues.